pkg/ocm/accounts_mgmt.go

```diff
@@ -81,30 +83,44 @@ func (a accountsMgmt) GetSubscription(ctx context.Context, subscriptionID strfmt
 	return responseVal, nil
 }
 
+func (a accountsMgmt) updateSubscription(ctx context.Context, subscriptionID strfmt.UUID, sub *amgmtv1.Subscription, err error) error {
+	if err != nil {
+		a.client.logger.Error(ctx, "Failed to create subscription request. Error: %v", err)
+		return err
+	}
+	return a.sendSubscriptionUpdateRequest(ctx, subscriptionID, sub)
+}
+
 // This function updates all the subscription's data that we didn't know at the subscription creation time
 // and we know only after installation is done such openshift_cluster_id.
 func (a accountsMgmt) UpdateSubscriptionPostInstallation(ctx context.Context, subscriptionID, openshiftClusterID strfmt.UUID) error {
 	defer commonutils.MeasureOperation("OCM-UpdateSubscriptionPostInstallation", a.client.log, a.client.metricsApi)()
 
 	sub, err := amgmtv1.NewSubscription().ExternalClusterID(openshiftClusterID.String()).Status(SubscriptionStatusActive).Build()
-	if err != nil {
-		a.client.logger.Error(ctx, "Failed to create subscription request. Error: %v", err)
-		return err
-	}
-
-	return a.sendSubscriptionUpdateRequest(ctx, subscriptionID, sub)
+	return a.updateSubscription(ctx, subscriptionID, sub, err)
 }
 
 func (a accountsMgmt) UpdateSubscriptionDisplayName(ctx context.Context, subscriptionID strfmt.UUID, displayName string) error {
 	defer commonutils.MeasureOperation("OCM-UpdateSubscriptionDisplayName", a.client.log, a.client.metricsApi)()
 
 	sub, err := amgmtv1.NewSubscription().DisplayName(displayName).Build()
-	if err != nil {
-		a.client.logger.Error(ctx, "Failed to create subscription request. Error: %v", err)
-		return err
+	return a.updateSubscription(ctx, subscriptionID, sub, err)
+}
+
+func (a accountsMgmt) UpdateSubscriptionConsoleUrlIfNeeded(ctx context.Context, subscriptionID strfmt.UUID, consoleUrl string) error {
+
+	if a.isConsoleUrlSet {
+		return nil
 	}
+	defer commonutils.MeasureOperation("OCM-UpdateSubscriptionConsoleUrl", a.client.log, a.client.metricsApi)()
 
-	return a.sendSubscriptionUpdateRequest(ctx, subscriptionID, sub)
+	sub, err := amgmtv1.NewSubscription().ConsoleURL(consoleUrl).Build()
+	err = a.updateSubscription(ctx, subscriptionID, sub, err)
+	if err == nil {
+		a.isConsoleUrlSet = true
+		a.client.logger.Info(ctx, "Updated console-url in subscription %s", subscriptionID)
```

Comment:
maybe we should add the cluster Id to the log

Reply:
We do not have access to the clusterID in that module, each cluster have a `ams_subscription` field in the DB so the connection can be done.

Do you think I need to add the clusterID to all API methods just for better logs?
@gamli75

Reply:
I think that we have it in the context - can you please verify it in the logs?

Reply:
OK, it should be in the logger from context.

It is missing in some parts because not all APIs contains the cluster_id in the request (`RegisterCluster` doesn't for example).

Opened a bug about the missing places https://issues.redhat.com/browse/MGMT-4749